Update details of Device-specific private key (#3643)
raspberrypi/documentation: documentation/asciidoc/computers/raspberry-pi/raspberry-pi-industrial.adoc

=== Device-specific private key

> Non-2712 devices have 8 rows of OTP (256 bits) available for use as a device-specific private key. On 2712 devices there are 16 rows of OTP data (512 bits). This is intended to support file-system encryption.

Reviewer: ```suggestion
Devices that use the Broadcom BCM2712 processor have 16 rows of OTP data (512 bits) to support filesystem encryption.
Devices that do not use BCM2712 have 8 rows of OTP (256 bits) available for use as a device-specific private key.
```
Suggestion: make it clearer that we're referring to the Broadcom BCM2712 processor.

Reviewer: _technically_ we could release a future device with a processor older than BCM2712, so I don't know if this necessarily holds up. Maybe just "Raspberry Pi 5" and "prior to Raspberry Pi 5"? But there are enough specifics here related to the CPU that this is presumably subject to change in future processors.

...definitely worth a revisit in the future when the situation changes. For now, I'm happy enough with it.

See https://gitlab.com/cryptsetup/cryptsetup[Cryptsetup] for more information about open-source disk encryption.

==== Key programming script `rpi-otp-private-key`

> The https://github.com/raspberrypi/rpi-eeprom/blob/master/tools/rpi-otp-private-key[`rpi-otp-private-key`] script wraps the device private key `vcmailbox` APIs in order to make it easier to read/write a key in the same format as OpenSSL.

Reviewer: ```suggestion
The https://github.com/raspberrypi/rpi-eeprom/blob/master/tools/rpi-otp-private-key[`rpi-otp-private-key`] script wraps the device private key `vcmailbox` APIs to make it easier to read and write a key in the OpenSSL format.
```

Suggestion: minor copy improvement.

Reviewer 2: " wraps the necessary `vcmailbox` APIs" ?

Reviewer 2: In fact I guess you could even link this to the `==== Mailbox API for reading/writing the key.` section?

> NOTE: The https://github.com/raspberrypi/usbboot[`usbboot`] repository contains all the tools you need and includes https://github.com/raspberrypi/rpi-eeprom[`rpi-eeprom`] as a git sub module.

Reviewer: ```suggestion
NOTE: The https://github.com/raspberrypi/usbboot[`usbboot`] repository contains all the tools you need, including https://github.com/raspberrypi/rpi-eeprom[`rpi-eeprom`] as a Git submodule.
```
Suggestion: Git uses 'submodule' as a single word.

f8dbc7b0a4fcfb1d706e298ac9d0485c2226ce8df7f7596ac77337bd09fbe160
----

> Writes a 32-byte randomly generated number to the device private key. +

Reviewer: ```suggestion
Writes a 32-byte randomly generated number to the device private key.

```
No need for the trailing `+` here; it actually prevents that warning from displaying as an admonition.

> WARNING: This operation cannot be undone. 

Reviewer: ```suggestion
WARNING: This operation cannot be undone.

```

For cleanliness' sake, we like to put newlines in between page elements.

[,bash]
----
# rpi-otp-private-key -w $(openssl rand -hex 32)
----

> NOTE: You can pass `-l <word count>` to specify the number of OTP rows to use and/or `-o <word offset>` to specify a start location in the key store.

Reviewer: ```suggestion
NOTE: To specify the number of OTP rows to use, pass `-l <word count>`. To specify a start location in the key store, pass `-o <word offset>`.
```
Suggestion: rephrase this to begin with the purpose, then explain the method.
